unittests/snapshot_tests.cpp

```diff
@@ -14,39 +15,52 @@ using namespace eosio;
 using namespace testing;
 using namespace chain;
 
-class snapshotted_tester : public base_tester {
-public:
-   snapshotted_tester(controller::config config, const snapshot_reader_ptr& snapshot, int ordinal) {
-      FC_ASSERT(config.blocks_dir.filename().generic_string() != "."
-         && config.state_dir.filename().generic_string() != ".", "invalid path names in controller::config");
+chainbase::bfs::path get_parent_path(chainbase::bfs::path blocks_dir, int ordinal) {
+   chainbase::bfs::path leaf_dir = blocks_dir.filename();
+   if (leaf_dir.generic_string() == std::string("blocks")) {
+      blocks_dir = blocks_dir.parent_path();
+      leaf_dir = blocks_dir.filename();
+      try {
+         auto ordinal_for_config = boost::lexical_cast<int>(leaf_dir.generic_string());
+         blocks_dir = blocks_dir.parent_path();
+      }
+      catch(const boost::bad_lexical_cast& ) {
+         // no extra ordinal directory added to path
+      }
+   }
+   return blocks_dir / std::to_string(ordinal);
+}
 
-      controller::config copied_config = config;
-      copied_config.blocks_dir =
-              config.blocks_dir.parent_path() / std::to_string(ordinal).append(config.blocks_dir.filename().generic_string());
-      copied_config.state_dir =
-              config.state_dir.parent_path() / std::to_string(ordinal).append(config.state_dir.filename().generic_string());
+controller::config copy_config(const controller::config& config, int ordinal) {
+   controller::config copied_config = config;
+   auto parent_path = get_parent_path(config.blocks_dir, ordinal);
+   copied_config.blocks_dir = parent_path / config.blocks_dir.filename().generic_string();
+   copied_config.state_dir = parent_path / config.state_dir.filename().generic_string();
+   return copied_config;
+}
 
-      init(copied_config, snapshot);
-   }
+controller::config copy_config_and_files(const controller::config& config, int ordinal) {
+   controller::config copied_config = copy_config(config, ordinal);
+   fc::create_directories(copied_config.blocks_dir);
+   fc::copy(config.blocks_dir / "blocks.log", copied_config.blocks_dir / "blocks.log");
+   fc::copy(config.blocks_dir / config::reversible_blocks_dir_name, copied_config.blocks_dir / config::reversible_blocks_dir_name );
```

Comment:
for blocks.index, it was not desired (unless Bart planned that), but the reindexing the block could be good and the only negatives is that that if there is an error creating the block index, it won't be very obvious.  So my preference is to leave it the same.

Reply:
Sounds good with leaving the blocks.index alone.

Regarding the reversible blocks dir, I think you will get some unlinkable block errors in the part of the test code you added that then tries syncing from LIB of the new chain to the head of the copied chain. (By the way, the fact that it was necessary to do that shows that the reversible blocks aren't being copied because it only replays the irreversible blocks in the blocks.log). I think it is okay to just leave it alone (for this PR).

Reply:
I'll write an Issue to add a test with copying over everything, but would rather not for this, since I need to rearrange the tests to shutdown the chain that is being copied, since it ends up with a dirty database right now.

Reply:
Yeah, I will add a PR to improve the tests after we get 2.0 out, but would rather leave these how they are for now.

Reply:
Sounds good.